tests/fdb/test_fdb_flush.py

```diff
@@ -0,0 +1,362 @@
+        ptfhost.template(src="fdb/files/fdb.j2", dest=self.FDB_INFO_FILE)
+
+    @pytest.fixture(scope="class", autouse=True)
+    def clearSonicFdbEntries(self, duthosts, rand_one_dut_hostname):
+        """
+            Clears SONiC FDB entries before and after test
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        duthost = duthosts[rand_one_dut_hostname]
+        duthost.shell(argv=["sonic-clear", "fdb", "all"])
+
+        yield
+
+        duthost.shell(argv=["sonic-clear", "fdb", "all"])
+
+    @pytest.fixture(scope="class", autouse=True)
+    def validateDummyMacAbsent(self, duthosts, rand_one_dut_hostname):
+        """
+            Validates that test/dummy MAC entry is absent before the test runs
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        duthost = duthosts[rand_one_dut_hostname]
+        pytest_assert(self.__getFdbTableCount(duthost, self.DUMMY_MAC_PREFIX) == 0, "Test dummy MAC is already present")
+
+    @pytest.fixture(scope="class", autouse=True)
+    def prepareDut(self, request, duthosts, rand_one_dut_hostname):
+        """
+            Prepare DUT for FDB test
+
+            It update the fdb_aging_time value, update the swss configuration, and restore SWSS configuration afer
+            test completes
+
+            Args:
+                request (Fixture): pytest request object
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        duthost = duthosts[rand_one_dut_hostname]
+        fdbAgingTime = request.config.getoption('--fdb_aging_time')
+
+        self.__deleteTmpSwitchConfig(duthost)
+        duthost.shell(argv=["docker", "cp", "swss:/etc/swss/config.d/switch.json", "/tmp"])
+        duthost.replace(
+            dest='/tmp/switch.json',
+            regexp='"fdb_aging_time": ".*"',
+            replace='"fdb_aging_time": "{0}"'.format(fdbAgingTime),
+            backup=True
+        )
+        duthost.shell(argv=["docker", "cp", "/tmp/switch.json", "swss:/etc/swss/config.d/switch.json"])
+        self.__loadSwssConfig(duthost)
+
+        yield
+
+        result = duthost.find(path=["/tmp"], patterns=["switch.json.*"])
+        if result["matched"] > 0:
+            duthost.shell(argv=["docker", "cp", result["files"][0]["path"], "swss:/etc/swss/config.d/switch.json"])
+            self.__loadSwssConfig(duthost)
+        self.__deleteTmpSwitchConfig(duthost)
+
+    def prepare_test(self, duthosts, rand_one_dut_hostname):
+        logging.info("Start prepare_test")
+
+        # Perform FDB clean up before each test
+        fdb_cleanup(duthosts, rand_one_dut_hostname)
+   
+        duthost = duthosts[rand_one_dut_hostname]
+
+        # save existing core files list
+        self.checkDutCorefiles(duthost, True)
```

Comment:
@lipxu Is it better to store existing core files at the very beginning? Maybe fdb cleanup operation can trigger core file.

Reply:
Since the cleanup operation before the test is not the DB operation, So I record the existing core file list after cleaning.
If cleanup operation triggers the core file, sanity check would catch such case.